### DISCUSSION TITLE
MPTCP Redundancy Mode

### DISCUSSION THREAD

**Discussion by aydinj** [Ysurac/openmptcprouter#3315] [Ideas]
I use MPTCP for online gaming in redundant mode to stabilize the lowest possible latency for UDP traffic and send the data replicated through all the active subflows available. I was using OpenMPTCProuter version 0.59.1-5.4. However, after upgrading to 0.60, the Multipath TCP scheduler doesn't have "redundancy." It is missed. Please add it.

**Comment by Ysurac** [comment]
With kernel 6.1, redundancy mode is no more available.
This will be added back in 6.6 kernel.

**Reply by pauleharris** [reply]
Hi,
sorry where in the UI do I find this in the 6.6 kernel?

**Reply by KevinashoUWU** [reply]
It is on network / MPTCP.
It's called red.o, something like that. But it doesn't work for me, it keeps doing bonding, could you try and see if it works for you?

**Reply by pauleharris** [reply]
okay
I tried bpf_red.o.  It is all over the map from a speed standpoint, 10 -> 250 in tests.  The failover aspect doesn't seem to work at all, it's the same responsiveness as bonding (5 seconds+ before any connectivity).

**Reply by KevinashoUWU** [reply]
Oh, it's the same thing that happens to me, it doesn't duplicate the packets, only bonding. Seamless failover doesn't work and it takes +5s to recover. I'll file a issue...

**Reply by pauleharris** [reply]
Hi,
Any updates or resolution here?  I've just reinstalled all the latest and still seeing this behavior.

**Reply by Ysurac** [reply]
Via SSH, what is the result of ```sysctl -a | grep mptcp``` ?

**Reply by pauleharris** [reply]
net.ipv4.tcp_available_ulp = espintcp mptcp tls
net.mptcp.add_addr_timeout = 120
net.mptcp.allow_join_initial_addr_port = 1
net.mptcp.checksum_enabled = 0
net.mptcp.enabled = 1
net.mptcp.pm_type = 0
net.mptcp.scheduler = default
net.mptcp.stale_loss_cnt = 4

**Reply by Ysurac** [reply]
Another scheduler is not used here. I will check what I get.

**Reply by Ysurac** [reply]
Issue should be fixed, new snapshot is compiling now.

**Reply by pauleharris** [reply]
thank you for the update.

I tried multiple images from here:
https://snapshots.openmptcprouter.com/6.6/ 
rpi4 and rpi5 - ext and squash.  

They seem to work, boot and run, then after attaching to the server in about 1 minute the rpi's loose all connectivity and I have to reflash.  I've updated the server, tried multiple etc.  
Thanks

**Comment by Bushingz** [comment]
Has this been resolved? My `sysctl -a | grep mptcp` shows this:
`
sysctl: error reading key 'net.ipv6.conf.all.stable_secret': I/O error
sysctl: error reading key 'net.ipv6.conf.default.stable_secret': I/O error
sysctl: error reading key 'net.ipv6.conf.erspan0.stable_secret': I/O error
sysctl: error reading key 'net.ipv6.conf.eth0.stable_secret': I/O error
sysctl: error reading key 'net.ipv6.conf.eth1.stable_secret': I/O error
sysctl: error reading key 'net.ipv6.conf.eth2.stable_secret': I/O error
sysctl: error reading key 'net.ipv6.conf.eth3.stable_secret': I/O error
sysctl: error reading key 'net.ipv6.conf.gretap0.stable_secret': I/O error
sysctl: error reading key 'net.ipv6.conf.ip6gre0.stable_secret': I/O error
sysctl: error reading key 'net.ipv6.conf.ip6tnl0.stable_secret': I/O error
sysctl: error reading key 'net.ipv6.conf.lo.stable_secret': I/O error
sysctl: error reading key 'net.ipv6.conf.sit0.stable_secret': I/O error
sysctl: error reading key 'net.ipv6.conf.teql0.stable_secret': I/O error
net.mptcp.add_addr_timeout = 120
net.mptcp.allow_join_initial_addr_port = 1
net.mptcp.checksum_enabled = 0
net.mptcp.enabled = 1
net.mptcp.pm_type = 0
net.mptcp.scheduler = default
net.mptcp.stale_loss_cnt = 4
`

Despite having selected redundancy:
<img width="297" alt="image" src="https://github.com/user-attachments/assets/6ea1bcf6-f1bb-482e-8491-2807e4945e2d" />

**Reply by Ysurac** [reply]
Did you try latest snapshot ?

**Reply by Bushingz** [reply]
I only used the 6.6 snapshot on the VPS, do I also need to use it on my router?

**Reply by Ysurac** [reply]
Yes

**Reply by Bushingz** [reply]
Thanks, that was it. It's fixed now.

**Comment by q2go** [comment]
On https://snapshots.openmptcprouter.com I see only snapshots for the router – where is the matching snapshot for the VPS?

**Reply by Ysurac** [reply]
https://github.com/Ysurac/openmptcprouter/wiki/Snapshots
